pyrsistent/_pmap.py

```diff
@@ -140,7 +140,9 @@ def __str__(self):
 
     def __hash__(self):
         # This hashing algorithm is probably not the speediest
```

Comment:
It's probably better to put it on the object itself than in some global cache.

I'm very interested in benchmarks based on real world experiences. I know that there are places that (I think) would be good candidates for optimisations but have not wanted to put too much effort into it before knowing that it's actually useful in the real world. Looking forward to it!